### ISSUE TITLE
Regarding the issue that the results of some stocks reproduced by the rule base strategy differ greatly from those in the paper

### ISSUE THREAD

**Issue by huangfeiyang** [waylonli/FINSABER#9]
Dear author, 
I have conducted a new round of reproduction this time and have some questions about the results. I would like to ask for your help to answer them.
1. During the process of reproducing finagent, a problem of missing libraries occurred. I checked the requirements and have not found the pandas_ta module so far. But I use pip install "git+https://github.com/aarigs/pandas-ta.git" command and it has been resolved.
2. When reproducing Table 2, I noticed that the results of the SMA Cross and WMA Cross strategies were very different from those in the paper. My command is as follows:
for s in \
BuyAndHoldStrategy \
SMACrossStrategy \
WMAStrategy \
ATRBandStrategy \
BollingerBandsStrategy \
TrendFollowingStrategy \
TurnOfTheMonthStrategy \
ARIMAPredictorStrategy \
XGBoostPredictorStrategy
do
MPLBACKEND=Agg python -m backtest.run_baselines_exp \
--setup cherry_pick_both_finmem \
--include "$s" \
--date_from 2022-10-06 \
--date_to 2023-04-10 \
--training_years 3 \
--rolling_window_size 1 \
--rolling_window_step 1
done
My experimental results are as follows:

<img width="1023" height="474" alt="Image" src="https://github.com/user-attachments/assets/57d20a1f-a085-44fd-9d5e-12d2c399a5d1" />
Paper results:

<img width="994" height="403" alt="Image" src="https://github.com/user-attachments/assets/d252fec4-4d8b-4536-9cf5-c8ce449b2930" />
My csv files:
[wma_results.csv](https://github.com/user-attachments/files/25621365/wma_results.csv)

[sma_results.csv](https://github.com/user-attachments/files/25621366/sma_results.csv)
I want to know the cause of this problem.
Thank you!

**Comment by huangfeiyang** [comment]
[wma_results.csv](https://github.com/user-attachments/files/25621381/wma_results.csv)
Here is the wma strategy result file.

**Comment by huangfeiyang** [comment]
> Regarding finrl, I would like to ask how we can ensure that transaction results are generated every time it is run. I conducted backtesting according to the window period of the paper. Many stocks did not generate transactions. I repeated the experiment several times in the same way. Stocks that were not traded before did generate transactions again, and the data was very close to that of the paper. So it's very hard for me to attribute the absence of transactions to randomness. I'm also curious about the cause of this issue. These are different results under two identical test commands: <img alt="Image" width="594" height="239" src="https://private-user-images.githubusercontent.com/26278538/556387952-9781caea-02bf-4204-b52a-77f584c5f22f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzIyODkwNDUsIm5iZiI6MTc3MjI4ODc0NSwicGF0aCI6Ii8yNjI3ODUzOC81NTYzODc5NTItOTc4MWNhZWEtMDJiZi00MjA0LWI1MmEtNzdmNTg0YzVmMjJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjI4VDE0MjU0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiNjNjMmNjOTRmZmVmY2ViZWRhMDRmOGEwNzAwOTc5ZDc4OTI1ZTIxNDRjMzAzMTdhZjViNTA5NDg1YzM4ZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.21zvI9GTSkmIeTzjWmxnUMjoi_hyWcmx9d_ggbNjO6Y">
> 
> <img alt="Image" width="912" height="186" src="https://private-user-images.githubusercontent.com/26278538/556388157-fc7c771e-bc7d-4d7c-98f2-d8cc2334d295.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzIyODkwNDUsIm5iZiI6MTc3MjI4ODc0NSwicGF0aCI6Ii8yNjI3ODUzOC81NTYzODgxNTctZmM3Yzc3MWUtYmM3ZC00ZDdjLTk4ZjItZDhjYzIzMzRkMjk1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMjI4VDE0MjU0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI1YzBlOGJmZDA5NjNmZmQxMzNhZWY4NTkyZmYyMjQ1MWYzZWY1YzFmY2IzNzBhNmZlNTNhZWZlMmY4MTcxNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DtMyXGc5Hk9Cj_ByhZU931uSTvmM6UC8yd8pjqrEWAw">

The command is as follows:
MPLBACKEND=Agg python -m backtest.run_baselines_exp \
  --setup cherry_pick_both_finmem \
  --include FinRLStrategy \
  --date_from 2022-10-06 \
  --date_to 2023-04-10 \
  --training_years 3 \
  --rolling_window_size 1 \
  --rolling_window_step 1

**Comment by waylonli** [comment]
Hi, thanks for the question. It looks like SMA and MMA are the only mismatches. The performance of these two strategies is highly sensitive to the `short_window` and `long_window` parameters specified in `backtest/strategy/timing/sma_crossover.py` and `backtest/strategy/timing/wma_crossover.py`. For our main experiments (Tables 3 and 4), these are set to (10, 20) by default. However, since the experiment for that table was run much earlier, we no longer have a record of the exact parameters used at the time. But I just managed to reproduce them using (20, 100), can you have a try and let me know you can get similar results?

Btw, we didn’t spend much effort tuning the parameters for the rule-based strategies, so feel free to experiment with them. Our primary goal is to validate the LLM traders’ performance, so there’s still plenty of room to improve the traditional methods through better tuning.

**Comment by huangfeiyang** [comment]
Thank you for your generous reply. I successfully reproduced the same results as in the paper following your prompts. Regarding the Finrl part, as I asked before, I still have some doubts about that issue. 
In addition, I would like to ask if there are any plans for this project to continue updating the data for 2025? If possible, I am also willing to actively participate in the work of data collection and cleaning! 
Once again, I extend my highest respect to you and this project. 
Thank you for your reply!

**Comment by waylonli** [comment]
Oh, sorry, I missed your question about FinRL. Yes, I’ve observed the same thing: the RL training can be a bit unstable. I tried modifying the reward to penalize cases where no trades are executed, but it ended up being too aggressive. To avoid manually introducing bias into the results, we ultimately decided not to make that change. That said, I’m not sure whether this issue is specific to the FinRL implementation, or if there are better packages available. I’d be happy to learn more if you have any suggestions.

Regarding the update plan: our research group does have more recent data, but it comes from private, paid datasets that can’t be made publicly available. In this repo, we’ve tried to use only public datasets so that everyone can experiment with it and run benchmarks. I think the biggest bottleneck is the news data, since FNSPID only provides news up to 2024.

If you’re interested in contributing to data collection and cleaning for a publicly available version, that would be extremely helpful. Please feel free to reach out to me privately with a brief introduction to yourself (you can find my email address on my [personal page](https://waylonli.com/)), and we can discuss the details there. I’ll close this issue for now, and I look forward to continuing the discussion.
